src/tagmanager/tm_workspace.c

```diff
@@ -577,6 +521,59 @@ gboolean tm_workspace_create_global_tags(const char *pre_process, const char **i
 	return ret;
 }
 
+/* Creates a list of global tags. Ideally, this should be created once during
+ installations so that all users can use the same file. This is because a full
+ scale global tag list can occupy several megabytes of disk space.
+ @param pre_process_cmd The pre-processing command. This is executed via system(),
+ so you can pass stuff like 'gcc -E -dD -P `gnome-config --cflags gnome`'.
+ @param sources Source files to process. Wildcards such as '/usr/include/a*.h'
+ are allowed.
+ @param tags_file The file where the tags will be stored.
+ @param lang The language to use for the tags file.
+ @return TRUE on success, FALSE on failure.
+*/
+gboolean tm_workspace_create_global_tags(const char *pre_process_cmd, const char **sources,
+	int sources_count, const char *tags_file, TMParserType lang)
+{
+	gboolean ret = FALSE;
+	GList *source_files = lookup_sources(sources, sources_count);
+
+	if (pre_process_cmd)
+		ret = create_global_tags_preprocessed(pre_process_cmd, source_files, tags_file, lang);
+	else
+	{
+		GList *node;
+		GPtrArray *filtered_tags;
+		GPtrArray *tags = g_ptr_array_new();
+		GSList *tm_source_files = NULL;
+
+		for (node = source_files; node; node = node->next)
```

Comment:
Could use `foreach_list()`

Reply:
`foreach_*()` and other Geany stuff isn't currently available in tag manager - it's still in the form of (kind of) external library and doesn't include anything from Geany.

Reply:
Oh, it's under src/ so I consider it part of Geany completely. But it's "your area" so OK

Reply:
It's not that I'm against it, it's just that nobody did it yet. If we decide to do that, we should probably do it properly and go through all the `for` loops and replace them if possible plus do other geany-style changes.